### ISSUE TITLE
Look into learning rate scheduler: once per batch or epoch?

### ISSUE THREAD

**Issue by EthanSansom** [EthanSansom/zooplankton#6]
Like the title, I'm not sure whether it's best for the scheduler to step once per batch or once per epoch. This might also depend on the scheduler type. Consider also a warm-up scheduler.

## Resources / Discussions

1. https://discuss.pytorch.org/t/scheduler-step-after-each-epoch-or-after-each-minibatch/111249
2. https://github.com/Tony-Y/pytorch_warmup/blob/master/examples/emnist/main.py

(1) suggests once per epoch, but (2) is once per batch.
